README.rst

```diff
@@ -1,114 +1,218 @@
+            <img width="10%" style="float: left;" src="https://raw.githubusercontent.com/unifyai/unifyai.github.io/master/img/externally_linked/logos/supported/numpy_logo.png">
         </a>
-        <img class="dark-light" width="6%" style="float: left;" src="https://raw.githubusercontent.com/unifyai/unifyai.github.io/master/img/externally_linked/logos/supported/empty.png">
     </div>
+    
+    <br clear="all" />
 
+------------------------------------------------------
+
+Ivy is both a ML transpiler and a framework, currently supporting JAX, TensorFlow, PyTorch and Numpy.
+
+Ivy unifies all ML frameworks 💥 enabling you not only to **write code that can be used with any of these frameworks as the backend**, 
+but also to **convert 🔄 any function written in any of them to code in your preferred framework!**
+
+You can check out `Ivy as a transpiler`_ and `Ivy as a framework`_ to learn more about this, try out Ivy
+straight away going through the `Setting up Ivy`_ section, or dive deep into Ivy's `Documentation`_ and `Examples`_!
+
+If you would like to contribute, you can join our growing `Community`_ 🌍, check out our `Contributing`_ guide,
+and take a look at the `open tasks`_ if you'd like to dive straight in 🧑‍💻 
+
+`lets-unify.ai <https://lets-unify.ai>`_ **together 🦾**
 
-.. _docs: https://lets-unify.ai/ivy
-.. _Colabs: https://drive.google.com/drive/folders/16Oeu25GrQsEJh8w2B0kSrD93w4cWjJAM?usp=sharing
-.. _`contributor guide`: https://lets-unify.ai/ivy/contributing.html
-.. _`open tasks`: https://lets-unify.ai/ivy/contributing/open_tasks.html
 
 Contents
 --------
 
-* `Overview`_
-* `Quick Start`_
-* `Background`_
-* `Design`_
-* `Extensions`_
+* `Ivy as a transpiler`_
+* `Ivy as a framework`_
+* `Setting up Ivy`_
+* `Documentation`_
+* `Examples`_
 * `Contributing`_
+* `Community`_
+* `Citation`_
 
-Overview
---------
+Ivy as a transpiler
+-------------------
+
+Ivy's transpiler allow you to use code from any other framework (and soon, from any other version of the same framework!) in your own code with just one line of code. To do so, Ivy traces a computational graph and leverages Ivy's frontends and backends to link one framework to the other. 
+
+This way, Ivy makes all ML-related projects available for you, independently of the framework you want to use to research, develop, or deploy systems. Feel free to head over to the docs for the full API reference, but the main functions that you'd use probably are:
+
+.. code-block:: python
+
+    ivy.compile()     # Compiles a function into an efficient graph, removing Ivy's wrapping and redundant code
+    ivy.transpile()   # Converts framework specific code to a different framework
+    ivy.unify()       # Converts framework specific code to Ivy
+
+These functions can be used eagerly and lazily. If you pass the neccesary arguments, the function will be called instantly, otherwise, compilation/transpilation will happen the first time you invoke the function with the proper arguments.
```

Comment:
If you pass the necessary arguments for function tracing, the compilation/transpilation step will happen instantly (eagerly). Otherwise, the compilation/transpilation steps will happen only when the returned function is first invoked (lazily).